public/app/features/search/reducers/manageDashboards.ts

```diff
@@ -46,13 +46,18 @@ const reducer = (state: ManageDashboardsState, action: SearchAction) => {
         }),
       };
     case MOVE_ITEMS: {
-      const { dashboards, folder } = action.payload;
-      const uids = dashboards.map((d: DashboardSectionItem) => d.uid);
+      const dashboards: DashboardSectionItem[] = action.payload.dashboards;
+      const folder: DashboardSection = action.payload.folder;
+      const uids = dashboards.map(db => db.uid);
       return {
         ...state,
         results: state.results.map(result => {
           if (folder.id === result.id) {
-            return { ...result, items: [...result.items, ...dashboards] };
+            return {
+              ...result,
+              items: [...result.items, ...dashboards.map(db => ({ ...db, checked: false }))],
+              checked: false,
+            };
```

Comment:
If possible, try to refactor this reducer to `createSlice` from redux-toolkit instead. That way we can minimize the hierarchy of spread operators and just mutate the state instead as createSlice uses ImmerJs.

Reply:
That's what I'd normally do for Redux reducers but this reducer for for React's `useReducer` hook, no Redux involved, so I don't think `createSlice` would work here.

Reply:
Ok I see, it seems like `useReducer` hook comes with some drawbacks like untyped dispatches and deep nested object spreading? 

You can still use Redux from functional components with `useState` hook so I'm not sure I like this pattern that much, but I'm probably not understanding the complete picture here. WDYT @dprokop @torkelo?

Reply:
On a fence here TBH. I totally see redux-toolkit is more mature and has a very elegant way to manage state, but moving state to Redux in case of search means quite some refactoring. Taking into consideration the fact that we wanted to move more complex state handling into the Redux store, I think search is a very solid candidate to migrate. Question is- do @Clarity-89 have time to do this now, maybe this could be scheduled for 7.1 enterprise planning?

Reply:
I agree, this will take some rework so better migrate this at a later stage.

Reply:
Didn't know we had a preference to use Redux even for local component state, but I'll keep that in mind in the future. 

Not sure about 7.1, seems like there are already other things planned, but will definitely look into this more later, especially if it starts becoming a bottleneck of some sort.